src/main/java/components/Storage.java

```diff
@@ -0,0 +1,312 @@
+		this.filePath = filePath;
+	}
+	
+	/**
+	 * This method loads the saved tasks list from the saved file into the program.
+	 *
+	 * @return the array list of tasks that keep tracks of the tasks.
+	 */
+	public ArrayList<Task> load() {
+		return this.readStored();
+	}
+	
+	/**
+	 * This method stores the to do task keyed in by the user into the file.
+	 *
+	 * @param input the description of the to do task
+	 * @throws EmptyDescriptionException if the description is empty for the to do task.
+	 */
+	public void storeTodo(String input) throws EmptyDescriptionException {
+		try {
+			BufferedWriter bw = new BufferedWriter(new FileWriter(filePath, true));
+			if (!input.substring(4).isEmpty()) {
+				String description = input.substring(4);
+				Todo todo = new Todo(description);
+				bw.write("T | 0 | " + description);
+				bw.newLine();
+				bw.flush();
+				bw.close();
+			} else {
+				throw new EmptyDescriptionException("todo");
+			}
+		} catch (IOException e) {
+			System.out.println(e.getMessage());
+		}
+	}
+	
+	/**
+	 * This method stores the deadline task keyed in by the user into the file.
+	 *
+	 * @param input the description of the deadline task
+	 * @throws EmptyDescriptionException if the description is empty for the deadline task.
+	 * @throws InvalidDescriptionException if the format of the deadline keyed in by the user is wrong.
+	 * @throws ParseException if the date format for the deadline keyed in by the user is wrong.
+	 */
+	public void storeDeadline(String input) throws EmptyDescriptionException, InvalidDescriptionException, ParseException{
+		try {
+			FileWriter fw = new FileWriter(this.filePath, true);
+			BufferedWriter bw = new BufferedWriter(fw);
+			if (input.contains("/by")) {
+				int index = input.lastIndexOf("/by");
+				String description = input.substring(8, index);
+				String by = input.substring(index + 3);
+				Date byDeadline = convertStringToDate(by);
+				if (description.isBlank()) {
+					throw new EmptyDescriptionException("deadline");
+				}
+				if (by.isBlank()) {
+					throw new InvalidDescriptionException("deadline");
+				}
+				Deadline deadline = new Deadline(description, byDeadline);
+				bw.write("D | 0 | " + description + " | " + by);
+				bw.newLine();
+				bw.flush();
+				bw.close();
+			} else {
+				throw new InvalidDescriptionException("deadline");
+			}
+		} catch (IOException e) {
+			System.out.println(e.getMessage());
+		}
+	}
+	
+	/**
+	 * This method stores the event task keyed in by the user into the file.
+	 *
+	 * @param input the description of the event task
+	 * @throws EmptyDescriptionException if the description is empty for the event task.
+	 * @throws InvalidDescriptionException if the format of the event keyed in by the user is wrong.
+	 * @throws ParseException if the date format for the event keyed in by the user is wrong.
+	 */
+	public void storeEvent(String input) throws EmptyDescriptionException, InvalidDescriptionException, ParseException{
```

Comment:
Maybe you could let your exceptions extend DukeException then you wouldn't have to specify so many exceptions